src/seer_server.erl

```diff
@@ -0,0 +1,166 @@
+-module(seer_server).
+
+-include("seer.hrl").
+
+-behaviour(gen_server).
+
+-export([init/1, handle_cast/2, handle_call/3, handle_info/2, terminate/2, code_change/3]).
+-export([start_link/0]).
+
+-record(state,
+        {mode :: undefined | stdout | carbon | carbon_offline | carbon_reconnected,
+         prefix :: undefined | binary(),
+         host :: undefined | binary(),
+         tcp_socket :: undefined | gen_tcp:socket(),
+         poll_interval :: undefined | non_neg_integer(),
+         carbon_buffer = [] :: [carbon_batch()],
+         buffer_size = 0 :: non_neg_integer(),
+         max_buffer_size :: non_neg_integer()}).
+
+-type state() :: #state{}.
+
+start_link() ->
+    gen_server:start_link({local, ?MODULE}, ?MODULE, [], []).
+
+init(_) ->
+    {ok, Host} = inet:gethostname(),
+    BinHost = case unicode:characters_to_binary(Host, latin1) of
+                Name when is_binary(Name) ->
+                    Name
+              end,
+    InitialState = #state{mode = ?ENV(?ENV_MODE, ?DEFAULT_MODE),
+                          prefix = ?ENV(?ENV_PREFIX, ?DEFAULT_PREFIX),
+                          host = BinHost,
+                          poll_interval = ?ENV(?ENV_POLL_INTERVAL, ?DEFAULT_POLL_INTERVAL),
+                          max_buffer_size = ?ENV(?ENV_MAX_BUFFER_SIZE, ?DEFAULT_MAX_BUFFER_SIZE)},
+    self() ! setup,
+    {ok, InitialState, 0}.
```

Comment:
Why are you returning a `Timeout` value of 0 here?

Reply:
I was under the impression that this was the way to specify no timeout, but reading the doc I see I can simply ignore that field completely.